posts/2023-01-16-how-prevent-computer-overload-remote-kind-clusters.md

```diff
@@ -0,0 +1,124 @@
+coverImageAlt: cool k8s related jpg
+---
+
+<p><a href="/topics/kubernetes">Kubernetes</a> can require a lot of resources, which can overload a developer's laptop. This article shows you how to use a set of tools—including <a href="https://kind.sigs.k8s.io/">kind</a>, <a href="https://kubernetes.io/docs/concepts/configuration/organize-cluster-access-kubeconfig/">kubeconfig</a>, and <a href="https://podman.io/">Podman</a> or <a href="https://www.docker.com/">Docker</a>—to spread your files around remote systems in support of your local development work.</p>
+
+<h2>Why I researched tools to prevent computer overload</h2>
+
+<p>Lately, I've been working a lot with <a href="https://open-cluster-management.io/">Open Cluster Management</a>, a community-driven project focused on multicluster and multi-cloud scenarios for Kubenetes applications.</p>
+
+<p>The Open Cluster Management topology is <em>hub-spoke</em> based, calling for one <em>hub</em> cluster and at least one <em>spoke</em> cluster. That means that, throughout my work, I needed at least two clusters running simultaneously.</p>
+
+<p>The quickest way to get two clusters up and running for development purposes is to use&nbsp;<a href="https://kind.sigs.k8s.io/">kind</a>, a Kubernetes management tool. With kind, you can easily spin up Kubernetes clusters running in containers on your local computer.</p>
+
+<p>One of my tasks included working with Prometheus, so I needed multiple clusters running the <a href="https://github.com/prometheus-operator/kube-prometheus">Prometheus operator</a> plus the required operators for running Open Cluster Management, including the <a href="https://open-cluster-management.io/getting-started/integration/app-lifecycle/">Application Lifecycle Manager</a> addon. The load on my local computer was, eventually, too much for it to handle, and it eventually stopped cooperating with me.</p>
+
+<p>To work around this bottleneck, I decided to spread my kind clusters to multiple computers around the office, import their kubeconfig files to my local computer, and continue to work as if the clusters were local.</p>
+
+<p>Each of the remote computers needed to install kind as well as a container engine. To manage the containers I used Podman, but Docker should do just as well.</p>
+
+<p>For access to remote clusters, SSH is usually preferable, but any manner of getting access to them should suffice. After spinning up a kind cluster and exporting the relevant kubeconfig file, you will no longer need access to the remote clusters, with the exception of the designated port 6443 for access to the Kubernetes API server.</p>
+
+<h2>How to set up&nbsp;kind remote clusters</h2>
+
+<p>The remote computer's IP address I use in the following examples is 192.168.1.102.</p>
+
+<p>Assuming you have SSH, connect to the remote computer as follows:</p>
+
+<pre>
+<code class="language-bash">$ ssh 192.168.1.102</code></pre>
```

Comment:
~~~suggestion
```bash
$ ssh 192.168.1.102
```
~~~
etc.

the rest of the HTML formatting can stay (better to use markdown, but NBD if it's a hassle), but these codeblocks should be converted to fenced markdown

Reply:
@bennypowers it might not be an hassle for this relatively small article.

but it is an hassle nevertheless.
I wanted this article to serve as an example for others who might want to copy over articles from other platforms but wish to avoid markdown adjustments.

Let's wait with this pr, I wanna look up a plugin that might add extra html tags rendering for our markdown generator.

Reply:
this is also a bigger hassle for more complex code snippets.
for example, in row 50, `&lt;&lt` will render as text using markdown.

Reply:
@bennypowers 
I couldn't find a quick solution, so I resolved this manually as you suggested in https://github.com/RedHat-Israel/RedHat-Israel.github.io/pull/58/commits/3fa6af9e295650dda5dab912803562c906904dd1.
But this is indeed a hassle, we should find a way around this or this will block potential articles from being added here.